core/src/main/java/com/sequenceiq/cloudbreak/service/rdsconfig/DbUsernameConverterService.java

```diff
@@ -38,7 +41,18 @@ public String toDatabaseUsername(String username) {
         return StringUtils.substringBefore(username, "@");
     }
 
+    private ConnectionNameFormat getConnectionNameFormat(DatabaseServerV4Response databaseServerV4Response) {
+        if (databaseServerV4Response.getDatabasePropertiesV4Response() != null
+                && databaseServerV4Response.getDatabasePropertiesV4Response().getConnectionNameFormat() != null) {
+            return databaseServerV4Response.getDatabasePropertiesV4Response().getConnectionNameFormat();
+        } else if (isAzure(databaseServerV4Response.getHost())) {
+            return ConnectionNameFormat.USERNAME_WITH_HOSTNAME;
+        } else {
+            return ConnectionNameFormat.SIMPLE;
+        }
+    }
+
     private static boolean isAzure(String hostname) {
```

Comment:
`isAzureSingleServer()`? I presume the `@hostname` part is not needed anymore for the Flexible Server.

Reply:
This part is for backward compatibilty, if redbeams doesn't provide the ConnectionNameFormat. Based on hostname we cannot decide if the db server is flexible or not. In this case we assume that the server is single if the hostname is matching with the azure hostname pattern. I can rename the method to describe our assumption better.